packages/video_player/closed_caption_file/lib/src/sub_rip.dart

```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import '../closed_caption_file.dart';
+
+/// This is a method that takes in a string (in the SubRip file format) and
+/// produces a list of [Caption]s.
+///
+/// This method should not be used directly, instead, use
+/// `ClosedCaptionFile.fromSubRipFile(...)`.
+List<Caption> parseCaptionsFromSubRipString(String file) {
+  final captions = <Caption>[];
+  for (final captionLines in _readSubRipFile(file)) {
+    if (captionLines.length < 3) break;
+
+    final captionNumber = int.parse(captionLines[0]);
+    final startAndEnd = _StartAndEnd.fromSubRipString(captionLines[1]);
+
+    final text = captionLines.sublist(2).join('\n');
+
+    final newCaption = Caption(
+      number: captionNumber,
+      start: startAndEnd.start,
+      end: startAndEnd.end,
+      text: text,
+    );
+
+    captions.add(newCaption);
+  }
+
+  return captions;
+}
+
+class _StartAndEnd {
+  final Duration start;
+  final Duration end;
+
+  _StartAndEnd(this.start, this.end);
+
+  // Assumes format from an SubRip file.
+  // For example:
+  // 00:01:54,724 --> 00:01:56,760
+  static _StartAndEnd fromSubRipString(String line) {
+    final times = line.split(' --> ');
```

Comment:
Good point. Added regex match.

Rather than make the start and end time 0, I'm opting to _not_ add the caption if it doesn't have a start or end time.